ipfs-cluster-service/main.go

```diff
@@ -290,6 +343,27 @@ func daemon(c *cli.Context) error {
 	checkErr("creating IPFS Connector component", err)
 
 	state := mapstate.NewMapState()
+
+	r, err := raft.ExistingStateReader(consensusCfg) // Note direct dependence on raft here
```

Comment:
This would be slightly different (See comment above). I think we can move most of this logic to a function too.